From 092ca6bbdd4885e1c725c8599f54d27b5584d4cc Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Mon, 5 Jun 2023 10:01:27 +0200
Subject: [PATCH] Run CRC32 checks on kernel image and fdt

Upstream-status: Inappropriate [configuration]
Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 include/configs/compulab-imx8m-plus.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/include/configs/compulab-imx8m-plus.h b/include/configs/compulab-imx8m-plus.h
index 2acf26ea6f..f2fcc37115 100644
--- a/include/configs/compulab-imx8m-plus.h
+++ b/include/configs/compulab-imx8m-plus.h
@@ -144,8 +144,8 @@
 		"usb_ul=usb start; setenv iface usb; setenv dev ${usb_dev}; setenv part ${boot_part};" \
 			"setenv bootargs console=${console} root=${usb_root} ${root_opt};\0" \
 		"ulbootscript=load ${iface} ${dev}:${part} ${loadaddr} ${script};\0" \
-		"ulimage=load ${iface} ${resin_dev_index}:${resin_root_part} ${zip_addr} boot/${image}; unzip ${zip_addr} ${loadaddr}; \0" \
-		"ulfdt=load ${iface} ${resin_dev_index}:${resin_root_part} ${fdt_addr} boot/${fdtfile};\0" \
+		"ulimage=load ${iface} ${resin_dev_index}:${resin_root_part} ${zip_addr} boot/${image}; unzip ${zip_addr} ${loadaddr};  run balena_kernel_load_crc_save; \0" \
+		"ulfdt=load ${iface} ${resin_dev_index}:${resin_root_part} ${fdt_addr} boot/${fdtfile}; run balena_fdt_load_crc_save; \0" \
 		"bootlist=sd_ul usb_ul emmc_ul\0" \
 	"bsp_bootcmd=echo Running BSP bootcmd ...; " \
 			"mmc dev 2; " \
@@ -153,6 +153,8 @@
 			"run resin_set_kernel_root; run set_os_cmdline; run emmc_ul;" \
 				"if run ulimage; then " \
 					"if run ulfdt; then " \
+						"run balena_kernel_load_crc_check;" \
+						"run balena_fdt_load_crc_check; " \
 						"booti ${loadaddr} - ${fdt_addr_r}; " \
 					"else " \
 						"if test ${boot_fdt} != yes; then " \
-- 
2.37.2

